src/tables.jl

```diff
@@ -11,6 +11,53 @@ function Base.getproperty(f::File, col::Symbol)
     end
 end
 
+# Column2
+ # copy
+ # BroadcastStyle/broadcasted
+Base.mapreduce(f, op, xs::Column2; kwargs...) = Base.mapfoldl(f, op, xs; kwargs...)
+
+@inline function Base.iterate(c::Column2)
+    length(c) == 0 && return nothing
+    array_lens = [length(x) for x in c.columns]
+    st = ThreadedIterationState(2, 1, 2, array_lens[1], array_lens)
+    return c.columns[1][1], st
+end
+
+@inline function Base.iterate(c::Column2, st)
+    row = st.row
+    array_index = st.array_index
+    array_i = st.array_i
+    row > length(c) && return nothing
+    if array_i + 1 > st.array_len
+        st.array_index += 1
+        st.array_i = 1
+        st.array_len = st.array_lens[min(end, st.array_index)]
+    else
+        st.array_i += 1
+    end
+    st.row += 1
+    return c.columns[array_index][array_i], st
+end
+
+Base.@propagate_inbounds function Base.getindex(c::Column2, i::Integer)
+    i′ = i
+    for C in c.columns
+        n = length(C)
+        i′ ≤ n && return C[i′]
+        i′ -= n
+    end
+    throw(BoundsError(c, i))
+end
+
+function Base.copy(c::Column2{T}) where {T}
```

Comment:
Couldn't this method be used also for `Column`?

Reply:
BTW, wouldn't it be faster to loop over `c.columns` and call `copyto!` on each? I guess you benchmarked that?

Reply:
Yeah, that would be faster. Actually, we could basically do a `memcpy` for most of the types which would be even faster.

Reply:
AFAIK `copyto!` does that for common array types already. Anyway that's the right level of abstraction for this kind of optimization.

Reply:
The problem is we're not dealing with `copyto!` from `Vector{T}` to `Vector{T}`, but from `Vector{UInt64}` to `Vector{T}`. For Int/Float64, the bits are the same, so we can manually `memcpy` the pointers.

Reply:
That sounds like an optimization that could be applied by Base too.